File: frontend/src/models/cpf.js
export default class CPF {
  regexCPF = /^\d{3}.\d{3}.\d{3}-\d{2}$/;

  constructor(cpf) {
    this.cpf = cpf;
  }

  isValid() {
    return this.regexCPF.test(this.cpf);
  }

  format() {
    this.cpf = this.cpf.replace(/\D/g, "");
    this.cpf = this.cpf.replace(/^(\d{3})(\d)/, "$1.$2");
    this.cpf = this.cpf.replace(/^(\d{3})\.(\d{3})(\d)/, "$1.$2.$3");
    this.cpf = this.cpf.replace(/\.(\d{3})(\d)/, ".$1-$2");
  }
}

File: backend/main_app/app/models/cpf.py
import re


class CPF:
    value = None
    __format = '^[0-9]{3}\.[0-9]{3}\.[0-9]{3}-[0-9]{2}$'

    def __init__(self, cpf: str):
        self.value = cpf

    def validate(self) -> bool:
        if self.value is None:
            return False
        else:
            return re.search(self.__format, self.value) is not None
